### PR TITLE
add addon:update and refactor addon code

### PR DESCRIPTION
**Pull request by wonderflow**

### CODE REVIEW

File: pkg/cmd/addon.go [hunk 1]
```diff
@@ -1,88 +1,108 @@
...
+		NewAddonConfigCommand(ioStream),
+		NewAddonListCommand(ioStream),
+		NewAddonUpdateCommand(ioStream),
 	)
 }
 
 func NewAddonConfigCommand(ioStreams cmdutil.IOStreams) *cobra.Command {
 	cmd := &cobra.Command{
-		Use:     "addon:config",
+		Use:     "addon:config <reponame> <url>",
```
Comment (zzxwill) [inline pkg/cmd/addon.go:30]: If we name every repository url, then we might need to `addon:switch` to switch the currently using repository. The name won't do more help to identify the url as the url itself would.

File: pkg/cmd/addon.go [hunk 2]
```diff
@@ -120,129 +150,33 @@ func NewAddonListCommand(c types.Args, ioStreams cmdutil.IOStreams) *cobra.Comma
...
-				}
-
-				if err != nil {
-					return err
-				}
-				//Check whether the definition is applied
-				var status = "uninstalled"
-				if _, err = cmdutil.GetWorkloadDefinitionByName(ctx, c, namespace, workloadDefinition.Name); err == nil {
-					status = "installed"
-				}
```
Comment (zzxwill) [inline pkg/cmd/addon.go:192]: Is it the right way to check the installation status? If yes, I can fulfill the TODO after the merge:)

File: pkg/cmd/refresh.go [hunk 3]
```diff
@@ -36,12 +35,6 @@ func NewRefreshCommand(c types.Args, ioStreams cmdutil.IOStreams) *cobra.Command
 	return cmd
 }
 
-func StatOrCreate(dir string) {
```
Comment (zzxwill) [inline pkg/cmd/refresh.go:39]: Cool.

File: pkg/cmd/addon.go (hunk 2 above)
Comment (wonderflow) [reply pkg/cmd/addon.go:192]: yes, but we should get definition from `.rudr/definitions/` instead of from cluster

File: pkg/cmd/addon.go (hunk 1 above)
Comment (wonderflow) [reply pkg/cmd/addon.go:30]: we can list them all, for example:

```
NAME          STATUS
repo1/route   installed
repo2/scaler uninstalled
```

```
vela addon:install repo2/scaler
```

**Pull request merged**